repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/regexrbac/RegexRbacAccessTest.groovy

```diff
@@ -0,0 +1,228 @@
+    }
+
+    @Unroll
+    def "a request for #resource which does not match any defined resource should be rejected"() {
+        when: "a request is made to Repose"
+        MessageChain messageChain = deproxy.makeRequest(
+            url: reposeEndpoint + resource,
+            method: GET,
+            headers: [(ROLES_HEADER): "any"]
+        )
+
+        then: "Repose responds with a 404"
+        messageChain.handlings.size() == 0
+        messageChain.receivedResponse.code as Integer == SC_NOT_FOUND
+
+        where:
+        resource << ["/url/casing", "/url/regex/12345", "/not/a/path"]
+    }
+
+    @Unroll
+    def "a request with #roles for #resource should succeed"() {
+        when: "a request is made to Repose"
+        MessageChain messageChain = deproxy.makeRequest(
+            url: reposeEndpoint + resource,
+            method: GET,
+            headers: roles ? [(ROLES_HEADER): roles] : []
+        )
+
+        then: "the origin service responds with a 200"
+        messageChain.handlings.size() == 1
+        messageChain.receivedResponse.code as Integer == SC_OK
+
+        where:
+        resource                    | roles
+        "/all"                      | null
+        "/any"                      | null
+        "/all"                      | "role1"
+        "/all"                      | "9elor"
+        "/any"                      | "role1"
+        "/any"                      | "9elor"
+        "/all"                      | "role1, role2"
+        "/any"                      | "role1, role2"
+        "/all/any"                  | "role1"
+        "/any/all"                  | "role1"
+        "/roles/simple"             | "role1"
+        "/roles/simple"             | "role2"
+        "/roles/simple"             | "role1,role2"
+        "/roles/simple"             | "role1, role2"
+        "/roles/nbsp"               | "nb sp"
+        "/roles/nbsp/end"           | "nbsp "
+        "/roles/casing"             | "ROLE1"
+        "/multimatch/one"           | "wildcard, multimatch"
+        "/multimatch/one/two/three" | "wildcard, multimatch"
+        "/inheritance/parent/child" | "child"
+        "/spaces"                   | "spaces"
+    }
+
+    @Unroll
+    def "a request with #roles for #resource should be rejected"() {
+        when: "a request is made to Repose"
+        MessageChain messageChain = deproxy.makeRequest(
+            url: reposeEndpoint + resource,
+            method: GET,
+            headers: [(ROLES_HEADER): roles]
+        )
+
+        then: "Repose responds with a 403"
+        messageChain.handlings.size() == 0
+        messageChain.receivedResponse.code as Integer == SC_FORBIDDEN
+
+        where:
+        resource                    | roles
+        "/roles/simple"             | "forbidden-role"
+        "/roles/casing"             | "role1"
+        "/roles/nbsp"               | "nb sp"
+        "/roles/nbsp/end"           | "nbsp "
```

Comment:
You can't tell the different, but these tests use the `&nbsp;` character rather than the space character. I just wanted to point these tests out because they make it apparent that we do not support roles which include the `&nbsp;` character. When a role is configured with a `&nbsp;`, it will _always_ be translated into a normal space.

Again, I'm not sure if that's a valid role in Identity, but it is something to be aware of.